src/main/java/com/liefery/android/icon_badge/IconBadge.java

```diff
@@ -198,21 +121,27 @@ public Bitmap export( int size ) {
     }
 
     private Bitmap renderBitmap( int size ) {
-        // The heightModifier mutates the height of the Bitmap to make space for
-        // the lower part of the Pin shape. Canvas drawing size is not changed
-        int heightModifier = ( backgroundShape == BACKGROUND_SHAPE_PIN ) ? Math
-                        .round( size * 1.5f ) : 0;
-
-        Bitmap bitmap = Bitmap.createBitmap( size + getShadowSizeX() * 2, size
-            + getShadowSizeY() * 2 + heightModifier, Bitmap.Config.ARGB_8888 );
-        int drawingAreaSize = bitmap.getWidth();
+        BackgroundProvider.Result result = backgroundProvider.export(size, 2 * elevation);
+        
+        Bitmap bitmap = Bitmap.createBitmap( (int)(result.width + (2 * elevation)), (int)(result.height + (2 * elevation)), Bitmap.Config.ARGB_8888 );
         Canvas canvas = new Canvas( bitmap );
-
-        draw( canvas, drawingAreaSize, size );
-
+    
+        drawShadow( canvas, result.path );
+        draw( canvas, size, result );
+        
         return bitmap;
     }
-
+    
+    private void drawShadow( Canvas canvas, Path shape ) {
+        shadowPaint.setShadowLayer(
+                elevation * 1.5f,
+                0f,
+                elevation,
+                Color.BLACK );
+        shadowPaint.setAlpha(70);
```

Comment:
Set alpha only once (e.g. in constructor) or ditch it all together and use a transparent color.